src/coreclr/jit/llvm.cpp

```diff
@@ -87,59 +139,156 @@ void EmitDoNothingCall(llvm::IRBuilder<>& builder)
     builder.CreateCall(_doNothingFunction);
 }
 
-bool visitNode(llvm::IRBuilder<> &builder, GenTree* node)
+Value* buildAdd(llvm::IRBuilder<>& builder, GenTree* node, Value* op1, Value* op2)
+{
+    if (!op1->getType()->isPointerTy() || !op2->getType()->isIntegerTy())
+    {
+        //only support gep
+        failFunctionCompilation();
+    }
+    return mapTreeIdValue(node->gtTreeID, builder.CreateGEP(op1, op2));
+}
+
+Value* buildCall(llvm::IRBuilder<>& builder, GenTree* node)
+{
+    GenTreeCall* gtCall = node->AsCall();
+    if (gtCall->gtCallType == CT_HELPER)
+    {
+        if (gtCall->gtCallMethHnd == _compiler->eeFindHelper(CORINFO_HELP_READYTORUN_STATIC_BASE))
+        {
+            const char* symbolName = (*_getMangledSymbolName)(_thisPtr, gtCall->gtEntryPoint.handle);
+            Function* llvmFunc = _module->getFunction(symbolName);
+            if (llvmFunc == nullptr)
+            {
+                llvmFunc = Function::Create(FunctionType::get(Type::getInt8PtrTy(_llvmContext), ArrayRef<Type*>(Type::getInt8PtrTy(_llvmContext)), false), Function::ExternalLinkage, 0U, symbolName, _module); // TODO: ExternalLinkage forced as linked from old module
+            }
+            // replacement for _info.compCompHnd->recordRelocation(nullptr, gtCall->gtEntryPoint.handle, IMAGE_REL_BASED_REL32);
+            (*_addCodeReloc)(_thisPtr, gtCall->gtEntryPoint.handle);
```

Comment:
I am not able to figure out what `addCodeReloc` does or why it is needed. Could you please shed some light on it for me?